### PR TITLE
Fix Snyk monitor bug

### PR DESCRIPTION
**Pull request by eslutz**
## Description

Updated Snyk monitor to use the `project.assets.json` file that should be in the same location regardless of if it is a single project repo or multi-project repo.

## Type of change: 2

1. Non-functional change (e.g. documentation changes, removing unused `using` directives, renaming local variables, etc)
2. Bug fix (non-breaking change that fixes an issue)
3. New feature (non-breaking change that adds functionality)
4. Breaking change (fix or feature that could cause existing functionality to not work as expected)

## Checklist:

- Have you reviewed your own code? Do you understand every change?
- Are you following the [contributing guidelines](../blob/main/CONTRIBUTING.md)?
- Have you added tests that prove your fix is effective or that this feature works?
- New and existing unit tests pass locally with these changes?
- Have you made corresponding changes to the documentation?
- Will this change require an update to an example project? (if so, create an issue and link to it)

---

<sup>_[Reviewer guidelines](../blob/main/CONTRIBUTING.md#reviewing-changes)_</sup>

### CODE REVIEW

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
         with:
           command: monitor
-          args: --file=${{ needs.set_variables.outputs.package_name }}/${{ needs.set_variables.outputs.package_name }}.sln
+          args: --file=${{ needs.set_variables.outputs.package_name }}/obj/project.assets.json
```
Comment (NeilSchiefer) [inline .github/workflows/build-and-publish.yml:92]: at this point is this a relative path?  If so do we know which directory we are in?

Comment (eslutz) [reply .github/workflows/build-and-publish.yml:92]: Yes, it's the root of the workspace that is uploaded as an artifact after the specified project it built.  So in this case the current workspace looks like this:
```bash
.
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING.md
├── Directory.Build.props
├── LICENSE.md
├── README.md
├── RockLib.DistributedTracing.AspNetCore
│   ├── AspNetCore
│   │   ├── CorrelationIdMiddleware.cs
│   │   ├── CorrelationIdMiddlewareExtensions.cs
│   │   ├── CorrelationIdMiddlewareOptions.cs
│   │   ├── HeaderNames.cs
│   │   └── HttpContextExtensions.cs
│   ├── CHANGELOG.md
│   ├── CorrelationIdAccessor.cs
│   ├── ICorrelationIdAccessor.cs
│   ├── RockLib.DistributedTracing.AspNetCore.csproj
│   ├── bin
│   │   └── Debug
│   └── obj
│       ├── Debug
│       ├── project.assets.json
│       └── project.nuget.cache
├── RockLib.DistributedTracing.sln
└── docs
```
With how the all the library repos are organized and how the publish pipeline works the `project.assets.json` will always be in that location to be used for the Snyk monitor job.

Comment (NeilSchiefer) [reply .github/workflows/build-and-publish.yml:92]: ah cool that should be good.  Ok can you explain the significance of that file argument then?  Does it matter that its the assets.json file vs the sln file?

Comment (eslutz) [reply .github/workflows/build-and-publish.yml:92]: No, for Snyk commands with .NET projects those are both options that can be used.  More details can be found in this Snyk documentation [Snyk For .NET](https://docs.snyk.io/products/snyk-open-source/language-and-package-manager-support/snyk-for-.net)

**Review by NeilSchiefer — APPROVED** [review]

**Review by motionsuggests — APPROVED** [review]

**Pull request merged**